### DISCUSSION TITLE
^M displayed while doing math calculation

### DISCUSSION THREAD

**Discussion by 0x8f701** [neovim/neovim#25453] [Q&A]
### Problem

While doing simple math of the following:

```
0 + 1
2 + 2
4 + 3
6 + 4
```

by following instructions of https://vimtricks.com/p/vim-calculator/, exceptly I use also macro to apply things to multiple lines

while using macro I got the following displayed in command 

```
=2 + 2^M4 + 3^M6 +
E488: Trailing characters: ^M4 + 3^M6 +
```

### Steps to reproduce

```
0 + 1
2 + 2
4 + 3
6 + 4
```

1. starts at the first line
2. qb start recording
3. v$y copy the content
4. `A = <c-r>= <c-r>"<cr>`
5. esc
6. q stop recording
7. `<c-v>` and select the following 3 lines
8. @b apply the macro

### Expected behavior

no ^M character in command

### Neovim version (nvim -v)

NVIM v0.10.0-dev

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Linux mint 21.1

### Terminal name/version

alacritty 0.11.0 (8dbaa0b)

### $TERM environment variable

xterm-256color

### Installation

apt-repository

**Comment by zeertzjq** [comment]
- Some characters seem to be missing from step 4.
- What keys (exactly) did you press in steps 7 and 8?

**Comment by wookayin** [comment]
The OP's reproduction step is different from what is explained in https://vimtricks.com/p/vim-calculator/.

I think the key sequence intended by OP is

```
qb
0
v$y
A = <C-r>=<c-r>"<CR>
<Esc>
q
j
@b
```

In step 4 `"` (i.e. pasting what was yanked int the `<C-R>` expression) is missing.

But this is not a valid because @GopherJ is using `v$y` instead of `vww` as explained in the referred doc -- the difference is that `v$y` yanks through the end of line which includes the EOL character (or line feed `^M`). So I think neovim is working as intended.

@GopherJ  You'll need to do `v$hy` or adjust the visual selection range properly so that it doesn't copy the `^M` character. Without that (e.g. change `v$y` to `v$hy`), the macro works as the you want.

**Comment by 0x8f701** [comment]
hi @wookayin yes it seem that I should avoid copy the last character, but I tested actually the macro works for a single line, but it doesn't work for multiple lines.

If I just apply it to single line it works, but if I use `<c-v>` to select multiple line, it doesn't work anymore and still report `^M` error

**Comment by 0x8f701** [comment]
I think the issue is more on how to apply this macro on multiple lines
